sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerAccessor.java

```diff
@@ -246,9 +249,21 @@ private static SpannerAccessor createAndConnect(SpannerConfig spannerConfig) {
     BatchClient batchClient =
         spanner.getBatchClient(DatabaseId.of(options.getProjectId(), instanceId, databaseId));
     DatabaseAdminClient databaseAdminClient = spanner.getDatabaseAdminClient();
+    String instanceConfigId = "unknown";
+    try {
+      instanceConfigId =
```

Comment:
Per doc spanner FQN has format `spanner:{projectId}.{instanceConfigId}.{instanceId}.{databaseId}.{assetId}`

However, currently instanceConfigId is not mentioned throughout SpannerIO. Had to make a real API call to get instance config Id.

Because spannerAccessor is stored in a static concurrent Map, this call will be done per (Dataflow worker) * (spanner instance destination)

cc: @manitgupta would like to learn if there a better way to do this.

Reply:
Do you know what is the avg and upper bound on "spanner instance destination" ?

Reply:
it is upper bounded by the number of SpannerIO (source+sink) used in the pipeline. (One SpannerIO transform only support single instance and database)